ttnn/cpp/ttnn/tensor/tensor_layout.hpp

```diff
@@ -0,0 +1,68 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "types.hpp"
+#include "enum_types.hpp"
+
+#include <ostream>
+
+namespace tt::tt_metal {
+
+class Size {
+public:
+    Size(size_t height, size_t width);
+    Size(const std::pair<size_t, size_t>& size);
+    Size(const std::array<size_t, 2>& size);
+
+    operator std::pair<size_t, size_t>() const;
+    operator std::array<size_t, 2>() const;
+
+    Size operator/(const Size& rhs) const;
+
+
+    // comparison operator
+    bool operator==(const Size& rhs) const;
+
+    size_t height() const;
+    size_t width() const;
+
+    // does not have to be a member, but it is easier to find if it is
+    Size aligned_to_tile(const Size& tile);
```

Comment:
Do you intend this class to represent more than just 2D shapes + some tile information? I think this class wouldn't serve too much purpose if just used (with some cleanup) the tile struct from metal.